trussworks/react-uswds
# feat: New Component SiteAlert

# Summary

This PR adds the [USWDS SiteAlert component](https://designsystem.digital.gov/components/site-alert). 

USWDS SiteAlert has four variants: `info`, `emergency`, `slim`, and `no-icon` that each apply different styles to the `<section>` root element. Brandon pointed out that these styles are not all mutually exclusive. 

In the pursuit of customizability, instead of including them all as possible values of the `variant` prop, two of the four are passed in as individual, optional properties: `showIcon` and `slim`. Separate props are preferred so that these styles can be combined with the two mutually exclusive variants, `info` and `emergency`, however the user wants.
 
The USWDS examples don't show all possible combinations, but I've included stories for the most likely combinations. In the last example, at Hana's suggestion, I used [Storybook Controls](https://storybook.js.org/docs/react/essentials/controls) to show the different applications  and combinations of styles with the `slim`, `showIcon`, and `variant` properties. If your control panel is hidden in Storybook, pressing the `D` key should make it reappear. Controls are only available for the last Storybook example.



## Related Issues or PRs

closes #981 


## How To Test

Check out this branch and run `yarn test` to execute tests and `yarn storybook` to see the new component in action. 

Alternatively, you can scroll to the bottom of this page and click "Show environments" on the right above the comment input box to access this component in Storybook. To see the different styles applied with Controls on the last story, press the `D` key in Storybook to access your controls panel and toggle the application of the `slim`, `showIcon`, and `variant` properties.
 

## Screenshots 

#### * Please note: I also thought the links were different colors because I accidentally took screenshots using one regular and one incognito browser window, but the colors are actually the result of USWDS styling

#### `info` links are blue, `emergency` links are grey

#### Standard SiteAlert, informational and emergency
![Screen Shot 2021-04-08 at 5 10 19 PM](https://user-images.githubusercontent.com/16230705/114110842-4d529300-988d-11eb-8ce8-5d754a47259b.png)


#### SiteAlert no header, informational and emergency
![Screen Shot 2021-04-08 at 5 10 40 PM](https://user-images.githubusercontent.com/16230705/114110860-59d6eb80-988d-11eb-8c4d-a16b591086d2.png)


#### SiteAlert with list, informational and emergency
![Screen Shot 2021-04-08 at 5 11 16 PM](https://user-images.githubusercontent.com/16230705/114110892-6f4c1580-988d-11eb-8042-d32d9c7a7425.png)



#### Slim Emergency Alert, Emergency Alert No Icon
![Screen Shot 2021-04-08 at 5 12 55 PM](https://user-images.githubusercontent.com/16230705/114110977-aa4e4900-988d-11eb-9588-5c901ed1b088.png)

#### SiteAlert With Custom Controls (default values)
![Screen Shot 2021-04-08 at 5 17 23 PM](https://user-images.githubusercontent.com/16230705/114111205-4a0bd700-988e-11eb-9f81-9d1eaf16d331.png)


#### SiteAlert With Custom Controls `slim`, `showIcon`, and `variant`
![Screen Shot 2021-04-08 at 5 16 20 PM](https://user-images.githubusercontent.com/16230705/114111164-247ecd80-988e-11eb-8d99-cd674ba06931.png)
